pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -151,8 +151,15 @@ func (c *Cluster) Start() error {
 			}
 			logger.Infof("deployment for mgr %s already exists. updating if needed", resourceName)
 
-			if err := updateDeploymentAndWait(c.context, c.clusterInfo, d, config.MgrType, mgrConfig.DaemonID, c.spec.SkipUpgradeChecks, false); err != nil {
-				logger.Errorf("failed to update mgr deployment %q. %v", resourceName, err)
+			ContIfUnhealthy := c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy
+
+			err := updateDeploymentAndWait(c.context, c.clusterInfo, d, config.MgrType, mgrConfig.DaemonID, c.spec.SkipUpgradeChecks, ContIfUnhealthy)
```

Comment:
I don't believe the `updateDeploymentAndWait()` method will behave any differently if this flag is passed, so we can revert this unless you find something in your testing or code analysis that shows it helps.
```suggestion
			err := updateDeploymentAndWait(c.context, c.clusterInfo, d, config.MgrType, mgrConfig.DaemonID, c.spec.SkipUpgradeChecks, false)
```

Reply:
@travisn  wrote:
> I don't believe the `updateDeploymentAndWait()` method will behave any differently if this flag is passed, so we can revert this unless you find something in your testing or code analysis that shows it helps.

I'll do some more code reading and testing and report back

Reply:
@obnoxxx  wrote:

> @travisn wrote:
> 
> > I don't believe the `updateDeploymentAndWait()` method will behave any differently if this flag is passed, so we can revert this unless you find something in your testing or code analysis that shows it helps.
> 
> I'll do some more code reading and testing and report back

According to what I see in the code, the `updateDeploymentAndWait()` method (we are useing the one from the mon here, invoked by mgr) does behave differently if a different value is passed here.

But I guess I haven't yet managed to test this thoroughly enough on my local system. Testing some more.

Reply:
I believe the different behavior only affects other daemon types, not the mgr.

Reply:
@travisn  wrote:
> I believe the different behavior only affects other daemon types, not the mgr.


well, several daemons use the mon variant of this function, including mgr:

```
$  git grep mon\.Update
pkg/operator/ceph/cluster/mgr/mgr.go:var updateDeploymentAndWait = mon.UpdateCephDeploymentAndWait
pkg/operator/ceph/cluster/rbd/mirror.go:var updateDeploymentAndWait = mon.UpdateCephDeploymentAndWait
pkg/operator/ceph/file/mds/mds.go:var UpdateDeploymentAndWait = mon.UpdateCephDeploymentAndWait
pkg/operator/ceph/file/mirror/mirror.go:var updateDeploymentAndWait = mon.UpdateCephDeploymentAndWait
pkg/operator/ceph/nfs/nfs.go:var updateDeploymentAndWait = opmon.UpdateCephDeploymentAndWait
pkg/operator/ceph/object/rgw.go:var updateDeploymentAndWait = mon.UpdateCephDeploymentAndWait
$